CHANGELOG.md

```diff
@@ -13,23 +13,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Optional `chem` dependencies are lazily imported
 - Order of README sections
+- Imports from top level `baybe.utils` no longer possible
```

Comment:
Perhaps we could use this opportunity to also rename `utils.numeric` to `utils.numerical` for concistency?

Reply:
done